src/encoding/hybrid_rle/mod.rs

```diff
@@ -25,6 +25,9 @@ enum State<'a> {
     None,
     Bitpacked(bitpacked::Decoder<'a, u32>),
     Rle(std::iter::Take<std::iter::Repeat<u32>>),
+    // Add a special branch for a single value to
```

Comment:
The new `Single` state that represent a single value. This branch is hot, hot hot!